pywhy_graphs/algorithms/generic.py

```diff
@@ -333,3 +334,212 @@ def _single_shortest_path_early_stop(G, firstlevel, paths, cutoff, join, valid_p
                     nextlevel[w] = 1
         level += 1
     return paths
+
+
+def _directed_sub_graph_parents(G, node):
+    """Finds the parents of a node in the directed subgraph.
+
+    Parameters
+    ----------
+    G : Graph
+        The graph.
+    node : node label
+        The node for which we have to find the parents.
+
+    Returns
+    -------
+    out : set
+        The parents of the provided node.
+    """
+    directed_parents = set(G.sub_directed_graph().predecessors(node))
+
+    return directed_parents
+
+
+def _bidirected_sub_graph_parents(G, node):
+    """Finds the parents of a node in the bidirected subgraph.
+
+    Parameters
+    ----------
+    G : Graph
+        The graph.
+    node : node label
+        The node for which we have to find the parents.
+
+    Returns
+    -------
+    out : set
+        The parents of the provided node.
+    """
+    bidirected_parents = set()
+
+    if not isinstance(G, CPDAG):
+        bidirected_parents = set(G.sub_bidirected_graph().neighbors(node))
+
+    return bidirected_parents
+
+
+def _is_collider(G, prev_node, cur_node, node):
+    """Checks if the given node is a collider or not.
+
+    Parameters
+    ----------
+    G : graph
+        The graph.
+    node : node
+        The node to be checked.
+
+    Returns
+    -------
+    iscollider : bool
+        Bool is set true if the node is a collider, false otherwise.
+    """
+    parents = _directed_sub_graph_parents(G, cur_node)
+    parents = parents.union(_bidirected_sub_graph_parents(G, cur_node))
+
+    if len(parents) == 2 and parents == {prev_node, node}:
```

Comment:
The number of parents does not need to be 2.

X -> Y <- Z with A -> Y

(X, Y, Z) is still a collider at Y.

Reply:
Yeah makes sense. I will add a test for it then patch it.